### ISSUE TITLE
Unification rules are not used anymore

### ISSUE THREAD

**Issue by QGarchery** [Deducteam/lambdapi#683]
It seems like unification hints are not used anymore by lambdapi since commit 5ca93cc96cc9d5795caf1c2ce4f5da0b199a3793 

The following example was checked before that and now returns `Some metavariables could not be solved, ...`
```
symbol Prop : TYPE;
symbol Type : TYPE;

symbol T : Type → TYPE;

symbol prop : Type;
rule T prop ↪ Prop;

symbol ∀ {x : Type} : (T x → Prop) → Prop;
notation ∀ quantifier;

unif_rule Prop ≡ T $t ↪ [ $t ≡ prop ];

symbol ⊥ : Prop ≔ `∀ C : Prop, C;
```

It type checks again if we replace the last line with: 
```symbol ⊥ : Prop ≔ @∀ prop (λ C : Prop, C); ```

**Comment by fblanqui** [comment]
Note that, if you declare T as injective, then the unification rule is useless.

**Comment by QGarchery** [comment]
This is awesome, together with fix of #649 ! Thanks !

I chose this rule for simplicity but I had another unification rule that was problematic and it looked like:
`unif_rule $a → $b ≡ T $c ↪ [ $a ≡ T $a'; $b ≡ T $b'; $c ≡ $a'  ⇒ $b' ];`

**Comment by QGarchery** [comment]
It works great ! I was explaining why I chose this unification rule in particular, sorry if it wasn't clear
